### ISSUE TITLE
Provide prebuilt release binaries for Nosey Parker

### ISSUE THREAD

**Issue by bradlarsen** [praetorian-inc/noseyparker#28]
Currently, Nosey Parker comes in source form, and has an automatically built Docker image available.

I'd like to provide a new distribution mechanism: a prebuilt binary distributed as a single file for popular platforms. To start with:

- GNU x86_64 Linux
- macOS x86_64
- macOS aarch64

The biggest obstacle to this currently is the use of Hyperscan in Rust:

- Hyperscan proper only builds on x86, but the Vectorscan fork of that project builds on other platforms (see #5)
- When installed from system package managers, static linking against Hyperscan/Vectorscan doesn't work as expected (there is still a dynamic dependency on `libhs`; see #22)
- Building Hyperscan or Vectorscan from source is nontrivial, requiring several additional dependencies (ragel, cmake, boost, ...)

**Comment by plygrnd** [comment]
➕ to this, the Docker container is quite slow on my M1 MacBook. A prebuilt binary would be excellent.

**Comment by bradlarsen** [comment]
Related: #39

**Comment by plygrnd** [comment]
Yup! I figured I'd build from source, then put the binary somewhere and
base a Homebrew formula off the binary (vs having Homebrew execute the
build process). What're your thoughts?

**Comment by bradlarsen** [comment]
Now in the latest on `main`, a vendored copy of Vectorscan is built from source. This makes the build process simpler (only CMake required beyond a Rust toolchain), and eliminates nonstandard runtime dependencies (like `libhs`). So it will now be feasible to simply copy around the appropriate `noseyparker` binary as a distribution mechanism.

In the next release, there will be a prebuilt `noseyparker` binary for `x86_64_unknown-linux-gnu`, and also a universal binary for macOS (usable on both x86_64 and ARM64). We have sorted out the macOS codesigning requirements and will have to take care of that once the release binary is ready.

There will _also_ be a multiplatform Docker image (#45) that natively supports both x86_64 and ARM64.

**Comment by bradlarsen** [comment]
This is done now! Release v0.13.0 includes prebuilt binaries for x86_64 Linux. It also includes a universal binary for both hardware flavors of Mac, signed and notarized with help from @Ash-Cat and @alexionescu.

**Comment by bradlarsen** [comment]
@dufferzafar thanks for reporting this! I opened a new issue (#58) for this. Let me look into how difficult it would be to link against `musl`.

In the meantime, you could build from source on that machine, which should eliminate the loader errors you are seeing.
